plugins/toolbox/src/components/Generators/CSR.tsx

```diff
@@ -0,0 +1,520 @@
+import React, { ChangeEvent, useEffect } from 'react';
+import TextField from '@mui/material/TextField';
+import Grid from '@mui/material/Grid';
+import Autocomplete from '@mui/material/Autocomplete';
+import FormControl from '@mui/material/FormControl';
+import ButtonGroup from '@mui/material/ButtonGroup';
+import Button from '@mui/material/Button';
+import { configApiRef, alertApiRef, useApi } from '@backstage/core-plugin-api';
+import { useToolboxTranslation } from '../../hooks';
+import { CopyToClipboardButton } from '../Buttons/CopyToClipboardButton';
+import { FileDownloadButton } from '../Buttons/FileDownloadButton';
+import PlayCircleOutlineRoundedIcon from '@mui/icons-material/PlayCircleOutlineRounded';
+import * as asn1js from 'asn1js';
+import {
+  CertificationRequest,
+  AttributeTypeAndValue,
+  RelativeDistinguishedNames,
+  GeneralNames,
+  GeneralName,
+  Extensions,
+  Extension,
+  Attribute,
+} from 'pkijs/build';
+import { getCrypto } from 'pkijs';
+
+const keyOptions = [
+  { label: 'RSA 4096', value: 'rsa-4096-sha2' },
+  { label: 'ECDSA p384', value: 'ecdsa-p384' },
+  { label: 'ECDSA p256', value: 'ecdsa-p256' },
+  { label: '⚠️RSA 2048 (SHA1 - Legacy systems only)', value: 'rsa-2048-sha1' },
+];
+
+export const CSRGenerator = () => {
+  const { t } = useToolboxTranslation();
+  const [fqdns, setFqdns] = React.useState<string[]>([]);
+  const [hashAlgName, setHashAlgName] = React.useState('SHA-256');
+  const [keyPEM, setKeyPEM] = React.useState('');
+  const [key, setKey] = React.useState<CryptoKeyPair | undefined>(undefined);
+  const [certReq, setCertReq] = React.useState('');
+  const [decodedCSR, setDecodedCSR] = React.useState('');
+  const [algorithmName, setAlgorithmName] = React.useState('rsa-4096-sha2');
+  const alertApi = useApi(alertApiRef);
+  const config = useApi(configApiRef);
+  const defaultSubject = config.getOptionalConfig('app.toolbox.csr.defaults');
+  const [subjectValues, setSubjectValues] = React.useState({
+    country: defaultSubject?.getString('country') || '',
```

Comment:
This probably should be `getOptionalString`?

Reply:
Additionally, you can just use `getOptionalString('toolbox.csr.defaults.country')`